docker/DOCKER.md

```diff
@@ -0,0 +1,72 @@
+# docker
+
+Docker images with balena-cli and docker-in-docker.
+
+## Architectures
+
+- `armv6`   (alpine only)
+- `armv7hf`
+- `aarch64` (debian only)
+- `amd64`
+- `i386`
+
+## Build
+
+```bash
+# optionally enable qemu for cross-compiling
+docker run --rm --privileged multiarch/qemu-user-static --reset -p yes
+
+export BALENA_ARCH="amd64"
+export BALENA_DISTRO="debian"
+export BALENA_CLI_VERSION="12.38.0"
+
+docker build ${BALENA_DISTRO} \
+    --build-arg BALENA_ARCH \
+    --build-arg BALENA_CLI_VERSION \
+    --tag "balenalib/${BALENA_ARCH}-${BALENA_DISTRO}-balenacli:${BALENA_CLI_VERSION}" \
+    --tag "balenalib/${BALENA_ARCH}-${BALENA_DISTRO}-balenacli:latest" \
+    --pull
+```
+
+## Usage
+
+### basic
+
+```bash
+# print usage
+docker run --rm -it balenalib/amd64-debian-balenacli:latest --help
```

Comment:
These examples all use `docker run` and don't include a command that requires logging in as a separate previous step. I understand that `docker run` creates a new container every time, which means that a token saved by `balena login` is not preserved for subsequent commands like `balena apps`. It also means that a docker daemon is started every time, which is verbose and slows things down. I know you already knew all of this and these are just examples, but I'm thinking that some inexperienced users would not know any better and would think that that's how it is. What about adding an example that runs a single container in the background (single `docker run` command, single docker daemon), then uses `docker exec` to run additional processes? For example:

```
$ docker run --detach --privileged --network host --name cli --rm -it balenalib/amd64-debian-balenacli /bin/bash

$ docker exec -it cli balena version -a
balena-cli version "12.38.1"
Node.js version "12.19.1"

$ docker exec -it cli balena login --token abc...

$ docker exec -it cli balena whoami
== ACCOUNT INFORMATION
USERNAME: ...
EMAIL:    ...
URL:      balena-cloud.com

$ docker exec -it cli balena apps
ID      APP NAME         SLUG                            DEVICE TYPE     ONLINE DEVICES DEVICE COUNT
1491721 test-nuc         gh_paulo_castro/test-nuc        intel-nuc       0              1
...

$ docker exec -it cli balena app test-nuc
== test-nuc
ID:          149...
DEVICE TYPE: intel-nuc
SLUG:        gh_.../test-nuc
COMMIT:      ce9...
```

Reply:
Hey, I agree with all your README feedback, it was meant as a placeholder/reference until we have official images and documentation, but there's a good chance external users will notice this so I'll give it a rework with your suggestions.

The examples I have here are trash, it was mostly so I could remember which flags are required for which type of action, but as you said these obviously aren't real use cases and would never work in stand-alone. I'll come up with some proper ones and get in the mindset of using this as a basis for official docs.